images/virtualization-artifact/pkg/controller/service/disk_service.go

```diff
@@ -66,21 +68,24 @@ func (s DiskService) Start(
 	source *cdiv1.DataVolumeSource,
 	obj ObjectKind,
 	sup *supplements.Generator,
-	wffc bool,
 ) error {
 	dvBuilder := kvbuilder.NewDV(sup.DataVolume())
 	dvBuilder.SetDataSource(source)
-	dvBuilder.SetPVC(storageClass, pvcSize)
 	dvBuilder.SetOwnerRef(obj, obj.GroupVersionKind())
-	dvBuilder.SetBindingMode(wffc)
 
-	// WaitForFirstConsumer is mainly needed for local storage.
-	// To prevent virtual machine migrations from failing, we set PVC to RWO so that virtual machines definitely cannot migrate.
-	if wffc {
-		dvBuilder.SetAccessMode(corev1.ReadWriteOnce)
+	sc, err := s.GetStorageClass(ctx, storageClass)
```

Comment:
If you only need the name, why not take it from the sprofile?

Reply:
GetStorageClass return default storage class if storageClass was not defined